modules/social_features/social_search/src/Plugin/search_api/processor/TaggingQuery.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Drupal\social_search\Plugin\search_api\processor;
+
+use Drupal\Core\Session\AccountInterface;
+use Drupal\search_api\LoggerTrait;
+use Drupal\search_api\Processor\ProcessorPluginBase;
+use Drupal\search_api\Query\QueryInterface;
+use Drupal\user\Entity\User;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Create a tagged conditions groups to make possible detect and alter them.
+ *
+ * @SearchApiProcessor(
+ *   id = "social_search_tagging_query",
+ *   label = @Translation("Tagging queries"),
+ *   description = @Translation("Tagging queries by entity type and entity type access."),
+ *   stages = {
+ *     "preprocess_query" = -30,
+ *   },
+ *   locked = true,
+ *   hidden = true,
```

Comment:
@nechai as we discussed would be good to make at least this one visible from UI.